enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/replication/id/ReplicatedIdAllocationStateMachine.java

```diff
@@ -43,24 +44,22 @@ public ReplicatedIdAllocationStateMachine( StateStorage<IdAllocationState> stora
     }
 
     @Override
-    public synchronized Optional<Result> applyCommand( ReplicatedIdAllocationRequest request, long commandIndex )
+    public synchronized void applyCommand( ReplicatedIdAllocationRequest request, long commandIndex,
+            Consumer<Result> callback )
     {
-        if( commandIndex <= state.logIndex() )
+        if ( commandIndex > state.logIndex() )
```

Comment:
minor: I really like the style where you check "acceptance criteria" first in a function and just return early if you don't pass, instead of wrapping the entire code block in a nested conditional.